ext/eglgles/gstegladaptation.c

```diff
@@ -584,35 +606,55 @@ gst_egl_adaptation_init_egl_surface (GstEglAdaptationContext * ctx,
       break;
     case GST_VIDEO_FORMAT_BGR:
     case GST_VIDEO_FORMAT_BGRx:
-    case GST_VIDEO_FORMAT_BGRA:
       frag_prog = g_strdup_printf (frag_REORDER_prog, 'b', 'g', 'r');
       free_frag_prog = TRUE;
       ctx->n_textures = 1;
       texnames[0] = "tex";
       break;
+    case GST_VIDEO_FORMAT_BGRA:
+      frag_prog = g_strdup_printf (frag_REORDER_full_prog, 'b', 'g', 'r', 'a');
+      free_frag_prog = TRUE;
+      ctx->n_textures = 1;
+      texnames[0] = "tex";
+      break;
     case GST_VIDEO_FORMAT_xRGB:
-    case GST_VIDEO_FORMAT_ARGB:
       frag_prog = g_strdup_printf (frag_REORDER_prog, 'g', 'b', 'a');
       free_frag_prog = TRUE;
       ctx->n_textures = 1;
       texnames[0] = "tex";
       break;
+    case GST_VIDEO_FORMAT_ARGB:
+      frag_prog = g_strdup_printf (frag_REORDER_full_prog, 'g', 'b', 'a', 'r');
```

Comment:
are you sure about 'g', 'b', 'a', 'r' ?

Reply:
I'm not, I just split the alpha based format maintaining the same logic as before

Reply:
Buffer:    ARGB
Shader:  RGBA
Then: R->G , G->B , B->A, A->R

It's correct